iroh/src/node/rpc.rs

```diff
@@ -131,92 +151,115 @@ impl<D: BaoStore> Handler<D> {
             BlobReadAt(msg) => chan.server_streaming(msg, self, Self::blob_read_at).await,
             BlobAddStream(msg) => chan.bidi_streaming(msg, self, Self::blob_add_stream).await,
             BlobAddStreamUpdate(_msg) => Err(RpcServerError::UnexpectedUpdateMessage),
+
             AuthorList(msg) => {
                 chan.server_streaming(msg, self, |handler, req| {
-                    handler.inner.docs.author_list(req)
+                    let (tx, rx) = flume::bounded(ITER_CHANNEL_CAP);
```

Comment:
So this used to return a stream, and now we give it a flume channel?

Can't we avoid this additional flume buffer by having something like

```rust
if docs.enabled {
  docs.author_list(...)
} else {
  stream::single(Err(anyhow!("docs are disabled")))
}
```

Somewhere in the various futures utils there used to be .left_stream() / .right_stream() to unify the types. Or box them in the worst case. Still better than another flume channel I think.

Reply:
We can't unless we box the streams, because otherwise the two branches have different types.
Maybe the boxing is fine. I am never sure when to try to avoid boxing and when not...?

Reply:
https://docs.rs/tokio-util/latest/tokio_util/either/enum.Either.html

Reply:
:+1: switched to this, much better